hanjaelee/app.js

```diff
@@ -52,20 +53,175 @@ app.post('/signup', (req, res) => {
       profile_image,
       password
     ) VALUES (?, ?, ?, ?);`
-  const rawData = database.query(rawQuery, [name, email, profileImage, password])
 
-  rawData
-    .then((data) => {
-      if (data.affectedRows) {
-        res.status(201).json({ message: 'userCreated' })
-        return
-      }
+  try {
+    const rawData = await database.query(rawQuery, [name, email, profileImage, password])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'userCreated' })
+    }
+    return res.status(202).json({ message: 'userNotCreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 3 게시글 등록하기
+app.post('/post', async (req, res) => {
+  const { title, content, userId } = req.body
+
+  const rawQuery =
+    `
+    INSERT INTO posts (
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);`
+
+  try {
+    const rawData = await database.query(rawQuery, [title, content, userId])
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'postCreated' })
+    }
+    return res.status(202).json({ message: 'postnotcreated' })
+  } catch (error) {
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
 
-      res.status(400).json({ message: 'userIsNotCreated' })
+// 과제 4 모든 게시물 가져오기
+app.get('/posts', async (req, res) => {
+  const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id
+    `
+  try {
+    await database.query(rawQuery, (err, rows) => {
+      const result = {}
+      const data = rows
+      result['data'] = data
+      res.status(200).json(result)
```

Comment:
await 키워드로 비동기 함수를 처리하고 있기 때문에 두번째 인자로 받는 콜백 함수 부분을 작성해줄 필요가 없어보입니다.
```suggestion
    const [result] = await database.query(rawQuery)
    
     return res.status(200).json(result)
```

Reply:
멘토님 피드백 주신대로 수정하면 result 에 row 가 하나만 들어간 구조가 되더라구요
이부분은 모든 게시물 가져오는 부분인데.. 추가 피드백 부탁드립니다!

Reply:
데이터가 여러개가 와야 하는데 피드백 주신대로 수정하면 데이터가 한개만 들어옵니다!
추가 피드백 부탁드리겠습니다!